server/webapp/WEB-INF/rails.new/app/controllers/admin/plugins/plugins_controller.rb

```diff
@@ -46,9 +47,35 @@ def upload
     end
   end
 
+  def edit_settings
+    plugin_settings = plugin_service.getPluginSettingsFor(params[:plugin_id])
+    render_settings_page(plugin_settings, 200)
+  end
+
+  def update_settings
+    plugin_settings = plugin_service.getPluginSettingsFor(params[:plugin_id], params[:plugin_settings])
+    plugin_service.validatePluginSettingsFor(plugin_settings)
+    if plugin_settings.hasErrors()
+      flash.now[:error] = l.string('SAVE_FAILED')
+      render_settings_page(plugin_settings, 400)
+    else
+      plugin_service.savePluginSettingsFor(plugin_settings)
+      render(:text => 'Saved successfully', :location => url_options_with_flash(l.string('SAVED_SUCCESSFULLY'), {:action => :index, :class => 'success'}))
```

Comment:
Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.